### ISSUE TITLE
Allow for different language SEPA emails

### ISSUE THREAD

**Issue by hannahswain** [woocommerce/woocommerce-gateway-stripe#744]
#### Affected ticket(s)

1465055-zen

#### Problem

SEPA emails are sent out in English, regardless of the site's language: 

```
Single Euro Payments Area

We're writing you to let you know that your purchase from [site name] for €1.10 will be debited from [bank account] within 2 days.

This transaction will appear on your bank statement as "[site name]", with mandate reference [reference] and creditor identifier [identifier]. 
You can reach out to [site name] with any questions. If you do not recognize this purchase, please respond to this email. 
Note: You are receiving this message from Stripe because [site name] uses Stripe to process payments online (https://stripe.com). 
..."
```

I know this has been raised multiple times before, and I had understood that Stripe's API didn't offer an option to adjust the language. 

This customer reached out to Stripe and let them know the following: 

> its available in the api To change this, a URL to view the mandate is returned as the value for sepa_debit[mandate_url]​ and the details are generated as part of the Source object creation[1]. So you would just need to change the URL to match the German text instead of the English one.

This is important for all of our international customers. Is this something we can implement?

-------------------

- [ ] Issue assigned to next milestone.
- [ ] Issue assigned a priority (will be assessed by maintainers).

**Comment by roykho** [comment]
I don't believe it goes by the site's language. Instead it should go with the customer's billing. Which it is already. We're passing customer's billing country here https://github.com/woocommerce/woocommerce-gateway-stripe/blob/master/includes/abstracts/abstract-wc-stripe-payment-gateway.php#L543

**Comment by hannahswain** [comment]
Okay, so we're doing it right, but Stripe still isn't picking it up and sending back the correct language emails?

**Comment by roykho** [comment]
I am not quite sure because I tried searching on their documentation to see how that is determined and didn't find anything useful.

**Comment by hannahswain** [comment]
@roykho we had a chat about this in the hallway - we agreed that you'd take a look and see if you can see why this isn't working. Thank you!

CC: @madeincosmos because you were also chatting with us

**Comment by roykho** [comment]
Yeah I looked into this and even specifically passed in a non US country and yet the email is still in English.

**Comment by hannahswain** [comment]
P2ing this so we can get the Partnerships team involved - it's important that we solve this, and I think we're going to need help from Stripe. 
https://woosupport.wordpress.com/2018/10/05/stripe-sepa-email-language/

**Comment by dougaitken** [comment]
tl;dr this will involved a significant engineering change from our side.

-----------------

Bringing @bmccotter's comment on the p2 (p6q7sZ-4Wj-p2) over here. 

```
If you’d like language options for recurring charge info in the email
itself, your team would need to write that. If you’d like to include a
link in an email that redirects the merchant to a Stripe mandate, that
mandate will have 7 language options to toggle between [1].

Your will need to update mandate[notification_method] shown here:
https://stripe.com/docs/sources/sepa-debit#using-webhooks-or-automated-emails

We provide language for mandates in other languages if you want to
customize emails:
https://stripe.com/docs/sources/sepa-debit#mandate-translations

Please feel free to reach out to priority-support@stripe.com and cc me

[1] https://stripe.com/sources/sepa_mandate?source=src_18HgGjHNCLa1Vra6Y9TIP6tU&client_secret=src_client_secret_XcBmS94nTg5o0xc9MSliSlDW
```

Roy asked for clarification if this is the initial email as he couldn't see how to change those. Britni confirmed with Stripe:

```
Hi Britni, correct, the standard email can’t be updated but you could
write and generate your own email using the language in our mandate
translations. If your engineering team has specific questions about
how to implement this our support team will be able to assist.
```

We still have 1465055-zen on hold so I'm going to let the user know we've logged this internally but it won't be any time soon, then mark as solved.

**Comment by thomasdiesenreiter** [comment]
Any news?

**Comment by jesscga** [comment]
This came up again in 3632380-zen - the customer contacted Stripe, who told them that the  `"preferred_locales" parameter needs to be switched to de-DE` -- I'm not sure if that's the same fix we had in mind or if it's referring to a different method, but wanted to post it in case it's helpful.

**Comment by bmccotter** [comment]
@v18 @thenbrent lets explore what it may take to support this.

**Comment by thenbrent** [comment]
It looks like we may just need to [set a customer's preferred locale(s)](https://stripe.com/docs/api/customers/create#create_customer-preferred_locales) based on this [doc](https://support.stripe.com/questions/language-options-for-customer-emails). That could be based on the WP site's locale at a minimum.

**Comment by thomasdiesenreiter** [comment]
(original bug reporter here =) )
Yes, please, it would be great it this could finally be fixed!
